### PR TITLE
[Chore] updated errorutil command doc documentation

### LINKED ISSUE #93: [Chore] errorutil: update documentation in command 'doc'
Update documentation to fix some errors and reflect recent discussions and updates to https://docs.meshery.io/project/contributing-error

### PR DESCRIPTION
**Pull request by mgfeller**
Signed-off-by: Michael Gfeller <mgfeller@mgfeller.net>

**Description**

This PR fixes #93 

**Notes for Reviewers**

No code changes, just documentation.

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by mgfeller** [comment]
@Shreyas220 and @sayantan1413 I've updated the doc in the utility based on the recent discussions we had. I would appreciate it if you could give it a quick look if you have time. Thanks!

File: cmd/errorutil/internal/coder/commands.go [hunk 1]
```diff
@@ -145,42 +145,48 @@ func commandDoc() *cobra.Command {
...
-  The file errorutil_analyze_summary.json contains a summary of the analysis, notably lists of duplicates etc.
-- The tool requires a file called component_info.json. Its location can be customized, by default it is the root directory (-d flag).
+
+This tool produces three files:
+- errorutil_analyze_errors.json: raw data with all errors and some metadata
+- errorutil_analyze_summary.json: summary of raw data, also used for validation and troubleshooting
+- errorutil_errors_export.json: export of errors which can be used to create the error code reference on the Meshery website
+
+Typically, the 'analyze' command of the tool is used by the developer to verify errors, i.e. that there are no duplicate names or details.
+A CI workflow is used to replace the placeholder code strings with integer code, and export errors. Using this export, a PR is created
```
Comment (Shreyas220) [inline cmd/errorutil/internal/coder/commands.go:174]: @mgfeller, I don't think the workflow creates a PR, it commits to master directly.

**Comment by sayantan1413** [comment]
Hey @mgfeller this is a really well written doc but I have a little suggestion like if we add that in the `errors.New(..)` we have to pass 5 parameters like 
- Code
- Short Description
- Long Description
- Probable Cause
- Suggested Remediation

**Comment by mgfeller** [comment]
> Hey @mgfeller this is a really well written doc but I have a little suggestion like if we add that in the `errors.New(..)` we have to pass 5 parameters like
> 
>     * Code
> 
>     * Short Description
> 
>     * Long Description
> 
>     * Probable Cause
> 
>     * Suggested Remediation

@Shreyas220 and @sayantan1413 thanks!. Good point, come to think of it (based on your feedback) I also think there should be some go-doc for this function where this can be elaborated as well.

**Comment by sayantan1413** [comment]
> > Hey @mgfeller this is a really well written doc but I have a little suggestion like if we add that in the `errors.New(..)` we have to pass 5 parameters like
> > ```
> > * Code
> > 
> > * Short Description
> > 
> > * Long Description
> > 
> > * Probable Cause
> > 
> > * Suggested Remediation
> > ```
> 
> @Shreyas220 and @sayantan1413 thanks!. Good point, come to think of it (based on your feedback) I also think there should be some go-doc for this function where this can be elaborated as well.

Yes @mgfeller that would be great 😃

**Comment by Shreyas220** [comment]
> > @Shreyas220 and @sayantan1413 thanks!. Good point, come to think of it (based on your feedback) I also think there should be some go-doc for this function where this can be elaborated as well.
> 
> Yes @mgfeller that would be great smiley

+1 :smiley:

**Comment by mgfeller** [comment]
@Shreyas220 @sayantan1413 I have pushed an updated and added some godoc. Wished godoc would support simple lists...

**Review by Shreyas220 — APPROVED** [review]
looks good to me 😄

**Review by sayantan1413 — APPROVED** [review]
Hey @mgfeller this look good to me as well. But I was thinking if we can also add that the first letter of each error description should be capitalized, I think this would have been better.

**Comment by mgfeller** [comment]
> Hey @mgfeller this look good to me as well. But I was thinking if we can also add that the first letter of each error description should be capitalized, I think this would have been better.

@sayantan1413 sure :+1:, I'll add that. Any other related conventions, like full stop at the end of each error description (or not)?

**Comment by sayantan1413** [comment]
No i dont think any other conventions are left 😅. I think we have covered all.

**Review by DelusionalOptimist — APPROVED** [review]
Looking great @mgfeller :+1:

**Pull request merged**